### PR TITLE
[IMP][13.0] stock: Add hook method

### PR DESCRIPTION
**Pull request by tranngocson1996**
I want to customize the value when creating the stock move but with the old code I can't do it.
So, I add you a hook method. Look it will be more professional and good for developer

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by tranngocson1996** [comment]
@rco-odoo @amoyaux plz review it !!! Thank

**Review by MiquelRForgeFlow — APPROVED** [review]

**Comment by tranngocson1996** [comment]
@MiquelRForgeFlow thank you my friend for approve. You can merge it 😄

**Comment by tranngocson1996** [comment]
@rco-odoo @amoyaux Ping !!!
cc: @Yenthe666

**Review by Yenthe666 — APPROVED** [review]

**Comment by Yenthe666** [comment]
I agree, one of those functions you just cannot get into right now  :)

**Comment by tranngocson1996** [comment]
@Yenthe666 thank you bro 😄. You can merge it 😄

**Comment by Yenthe666** [comment]
I think technically I can do it as I have more rights than almost everyone but I'm afraid I'm not allowed too sorry :) Has to be reviewed & merged by an employee from Odoo SA itself so let's wait for the stock guys.

**Comment by tranngocson1996** [comment]
@Yenthe666 ok bro, I will wait for them 😞

File: addons/stock/models/stock_picking.py [hunk 1]
```diff
@@ -693,6 +693,21 @@ def action_cancel(self):
         self.write({'is_locked': True})
         return True
 
+    def prepare_stock_move_vals(self, line, picking):
```
[the remark this answers is not in the document]
Comment (amoyaux) [reply addons/stock/models/stock_picking.py:696]: and not in the public method as defined in the guidelines

[the remark this answers is not in the document]
Comment (tranngocson1996) [reply addons/stock/models/stock_picking.py:696]: > and not in the public method as defined in the guidelines

@amoyaux before, I've set this method is private, but runbot build fail

[the remark this answers is not in the document]
Comment (tranngocson1996) [reply addons/stock/models/stock_picking.py]: Done. wait CI

**Comment by amoyaux** [comment]
robodoo r+

**Comment by amoyaux** [comment]
robodoo retry

**Comment by Yenthe666** [comment]
@mart-e @odony interesting behaviour from robodoo. @amoyaux did a retry on https://github.com/odoo/odoo/pull/84249#issuecomment-1041363928 and robodoo posts it here 😄

**Comment by MiquelRForgeFlow** [comment]
@Yenthe666 I think the correct ones to ping in this case should be @xmo-odoo or @Xavier-Do, as they are the guru maintainers of https://github.com/odoo/runbot

**Comment by Yenthe666** [comment]
> > interesting behaviour from robodoo. amoyaux did a retry on [#84249 (comment)](https://github.com/odoo/odoo/pull/84249#issuecomment-1041363928) and robodoo posts it here smile
> 
> Amoyaux actually performed a retry on this PR, then realised they were on the wrong PR and deleted the comment 13 seconds later.
> 
> Sadly for them the mergebot gets all its information through the webhooks so had already processed the command and doesn't take deletions in account, thus had already queued up the response you can see above.

Aha, interesting! Thanks for answering that :)

**Pull request closed without merging**